bitcoin/src/blockdata/locktime/relative.rs

```diff
@@ -270,47 +270,80 @@ impl fmt::Display for Time {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
 }
 
-/// Errors related to relative lock times.
+/// Input time in seconds was too large to be encoded to a 16 bit 512 second interval.
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[non_exhaustive]
-pub enum Error {
-    /// Input time in seconds was too large to be encoded to a 16 bit 512 second interval.
-    IntegerOverflow(u32),
-    /// Tried to satisfy a lock-by-blocktime lock using a height value.
-    IncompatibleHeight(LockTime, Height),
-    /// Tried to satisfy a lock-by-blockheight lock using a time value.
-    IncompatibleTime(LockTime, Time),
+pub struct TimeOverflowError {
+    /// Time value in seconds that overflowed.
+    // Private because we maintain an invariant that the `seconds` value does actually overflow.
+    pub(crate) seconds: u32
```

Comment:
its constructed in `blockdata::transaction`

Reply:
Ah, I didn't notice that *`Sequence`* is not in the same module. It begs the question whether it should be moved to relative module similarly to `absolute::LockTime` but maybe not because `Sequence` is more than lock time.

Reply:
Lets call that out of scope for this PR?  (FTR I always wondered why `Sequence` went were it went.)

Reply:
Agreed, out of scope for this PR.

But the reason `Sequence` is part of the transaction module is that sequence numbers are defined as part of transaction inputs.

But I'd be fine moving it into the units crate because it's (usually) interpreted as a relative locktime.